### PR TITLE
Rocketlet.json schema definition

### PR DESCRIPTION
**Pull request by graywolf336**
It's a work in progress that is highly subject to change, but committing it will allow me to reference it via other projects.

### CODE REVIEW

File: src/Rocketlet.ts [hunk 1]
```diff
@@ -9,14 +10,10 @@ export abstract class Rocketlet {
...
      */
-    protected constructor(private readonly name: string,
-                          private readonly id: number,
-                          private readonly version: string,
-                          private readonly description: string,
-                          private readonly requiredApiVersion: string,
-                          private readonly author: IRocketletAuthor) {
-        console.log(`Constructed the Rocketlet ${this.name} (${this.id})`,
-            `v${this.version} which depends on the API v${requiredApiVersion}! Created by ${author.name}`);
+    protected constructor(private readonly info: IRocketletInfo) {
```
Comment (graywolf336) [inline src/Rocketlet.ts]: `IRocketletInfo` will be passed into the Rocketlet's constructor which is parsed from the included `rocketlet.json` file.

Comment (mrsimpson) [inline src/Rocketlet.ts]: 👍 for having one structure for all metadata

File: src/accessors/IPersistenceRead.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { IRelatedPersistenceRead } from './IRelatedPersistenceRead';
  * Provides access to the database
  */
 
-export interface IPersistence {
+export interface IPersistenceRead {
```
Comment (mrsimpson) [inline src/accessors/IPersistenceRead.ts]: @graywolf336 any idea why this appears on the pull requests's change list?

Comment (graywolf336) [inline src/accessors/IPersistenceRead.ts]: It was a slip up in your pull request, was easier to merge as is and update my version.

**Pull request merged**